### ISSUE TITLE
Weird behaviour with combination of ShowEntire, MaxHeight and StopPaging

### ISSUE THREAD

**Issue by MercinaM** [QuestPDF/QuestPDF#776]
**Describe the bug**
I was editing an existing report, and trying to limit the height of the first text segment in the headers to two lines. As soon as I did that, the space for the header was still reserved, but the header itself was blank.

The fix itself turned out to be pretty easy, I just switched the last item from using "PaddingBottom(60)" to "Height(60)", which I probably should have been doing in the first place.

Still, since the height of the last item is the same in both cases (60), and the height of the header in its entirety is therefore also the same, I don't understand why there would be a difference.

Also of note, if I remove the "ShowEntire" call (which shouldn't make a difference in this example, since the contents fit on a single page anyway), the problem also goes away even if I keep using "PaddingBottom".

**To Reproduce**

```
Document
    .Create(container =>
    {
        container.Page(page =>
        {
            page.Margin(38);
            page.DefaultTextStyle(new TextStyle().FontFamily(Fonts.Verdana).FontSize(9));
            page.Size(PageSizes.A4);

            page.Header().AlignLeft().Element(container =>
            {
                container.ShowEntire().Column(column =>
                {
                    column.Item().SkipOnce().PaddingTop(1).BorderBottom(0.5f).BorderColor(Colors.Black);
                    column.Item().Element(item =>
                    {
                        item.Column(col =>
                        {
                            column.Item().MaxHeight(50).StopPaging().Text("000000001 000000002 000000003 000000004 000000005 000000006 000000007 000000008 000000009 000000010 000000011 000000012 000000013 000000014 000000015 000000016").FontSize(14);
                            column.Item().Text("line 1\r\nline 2\r\nline 3\r\nline 4\r\nline 5\r\nline 6").FontSize(7);
                        });

                    });

                    // Works when replaced with "Height"
                    column.Item().PaddingBottom(60);
                });
            });

            page.Content().Element(container => container.Height(10).Background(Colors.Black));
        });
    })
```

**Expected behavior**
The contents of the header should be rendered either way.

**Screenshots**

With "PaddingBottom":
![image](https://github.com/QuestPDF/QuestPDF/assets/12325231/4167f061-4e5e-4a82-adc5-d6809ba24bd2)

With "Height":
![image](https://github.com/QuestPDF/QuestPDF/assets/12325231/3b15489f-26e5-458d-a6b2-5f5953f88dd5)

**Environment**
Library version: 2023.12.4
OS: Windows 11, x64

**Comment by MarcinZiabek** [comment]
Thank you for sharing this problem!

I have investigated your code and found the root cause. It is related to the floating-point arithmetic and its rounding errors. I attempted to fix it with this https://github.com/QuestPDF/QuestPDF/commit/90541d42080a199f793b8b6e352b6c606dd3ffc5 commit. 

I plan to publish the `2023.12.4` release soon and also backport this fix to the `2022.12` pure-MIT release cycle as part of the extended support.
